lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,551 @@
+                      Traffic Manager allows endpoints in this profile to response to the health check.
+            tolerated_number_of_failures:
+                description:
+                    - The number of consecutive failed health check that Traffic Manager tolerates
+                      before declaring an endpoint in this profile Degraded after the next failed health check.
+    state:
+        description:
+            - Assert the state of the resource group. Use 'present' to create or update and
+              'absent' to delete. When 'absent' a resource group containing resources
+              will not be removed unless the force option is used.
+        default: present
+        choices:
+            - absent
+            - present
+        required: false
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+author:
+    - "Julio Colon (@code4clouds)"
+    - "Diego Casati (@diegocasati)"
+'''
+
+EXAMPLES = '''
+    - name: Create Traffic Manager Profile
+      azure_rm_trafficmanagerprofile:
+        name: "contoso.com"
+        state: "present"
+        resource_group: "ContosoRG"
+        properties:
+          profile_status: "Enabled"
+          traffic_routing_method: "Performance"
+          dns_config:
+            ttl: "300"
+          monitor_config:
+            protocol: "HTTP"
+            port: 80
+            path: "/monitor/index.html"
+            status: "Active"
+            interval_in_seconds: 30
+            timeout_in_seconds: 10
+            tolerated_number_of_failures: 3
+          location: "global"
+          tags:
+            project: "My Project"
+
+    - name: Delete a Traffic Manager Profile
+      azure_rm_trafficmanagerprofile:
+        name: "contoso.com"
+        state: "absent"
+'''
+
+RETURN = '''
+state:
+    description: Current state of the resource group.
+    returned: always
+    type: dict
+    sample: {
+        "id": "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/Testing",
+        "location": "westus",
+        "name": "Testing",
+        "provisioning_state": "Succeeded",
+        "tags": {
+            "delete": "on-exit",
+            "testing": "no"
+        }
+    }
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.trafficmanager.models import DnsConfig, Profile, MonitorConfig, Endpoint
+except ImportError:
+    pass
+
+
+def trafficmanagerprofile_group_to_dict(tm):
+    return dict(
+        id=tm.id,
+        name=tm.name,
+        resource_group=tm.resource_group,
+        location=tm.location,
+        tags=tm.tags)
+
+
+class AzureRMTrafficManager(AzureRMModuleBase):
+    # Main Traffic Manager class
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            name=dict(type='str', required=True),
+            resource_group=dict(type='str', required=True),
+            properties=dict(type='dict'),
```

Comment:
What are the properties? As other modules, we flatten the properties in the ansible module's parameter. And for the second level of dict type, we can use `options` to specific another spec to define the key-value pairs in the dict.